### ISSUE TITLE
How do I buy you a coffee?

### ISSUE THREAD

**Issue by papb** [taximanli/kotobaasobou#2]
❤️ 

This is fantastic work. Do you have any platform where I can donate to? You could enable GitHub sponsors, perhaps?

**Comment by taximanli** [comment]
Thanks for your compliment! I'm glad that you enjoyed the game. I tried enabling GitHub sponsor but the paperwork was quite tedious. I managed to set up a [ko-fi page](https://ko-fi.com/taximanli) for myself. I ❤️ coffee!

And I've developed a new version of the game and I'm planning to release it in couple of days. Please stay tuned.

**Comment by taximanli** [comment]
I've completed a new version of the game and it's now live! The game now supports IME keyboard input and includes Hint Mode which gives 3 extra types of hints to help players in guessing the word.

**Comment by jeffrey-friedl** [comment]
I hate the new version... it exposes my poor Japanese skill even more than the old version. 😂

Thanks so much for your work.  Three suggestions:

1) Add "あそぼう" as an allowed guess, just because.
2) Have "hint mode" be off by default.
3) In the 「type かな here」box, and/or the guess boxes prior to the guess being officially submitted, highlight (or de-highlight?) characters that have been shown to not be in the answer.  I would find this helpful, especially with the new ability to type the guess (thanks so much for that!)

Thanks!

**Comment by jeffrey-friedl** [comment]
"Hint mode" is off, but I'm seeing two different shades of green, and looking at the HTML, one css class is marked "close". Is that some kind of hint? Please allow the original, no-hint game play. 🙏

**Comment by taximanli** [comment]
> "Hint mode" is off, but I'm seeing two different shades of green, and looking at the HTML, one css class is marked "close". Is that some kind of hint? Please allow the original, no-hint game play. 🙏

Yes the lighter shade of green (lime colour) indicates the kana submitted in this spot is close to the correct kana (basic kana/voiced kana "dakuten"/semi-voiced kana "handakuten"/small kana). I'll explain with this example. Let's say the correct word of the day is はってん (development), and you submitted this guess はつでん (electricity generation). The tile colours will turn to GREEN-LIME-LIME-GREEN.

In the old version, to balance the difficulty of the guessing the word, the game ignored small kana (treated や, ゆ, よ, つ the same as ゃゅょっ), that was not truly reflecting the correct Japanese spelling. So in this new version, the game treats small kana as separate letters to their large counterparts, therefore LIME coloured tile is introduced to maintain the game difficulty roughly the same as the old version.

I also read feedback from players saying that the old version was too difficult because there were 70+ kana to pick as answer s and I think giving the LIME coloured tile hint for voiced kana "dakuten"/semi-voiced kana "handakuten" can make the game more playable?

Let me consider moving the LIME coloured tile hint into Hint Mode so when Hint Mode is switched off, the new version will be at the more or less the same level of difficulty as the old version.

**Comment by jeffrey-friedl** [comment]
I eventually loaded the page in a new browser, and because I had no cookies, the instructions were presented and understood what the subtle color difference meant.  During gameplay, I don't know that I would be able to tell which shade of green a single green square was; if two colors showed up, I'd be able to tell the difference, of course.

Personally,  I very much don't want these hints, but I understand how some may want them. (Frankly, I'd rather have more guesses; if it ends up taking 15 guesses, anyone who thinks the limit should be 12 will know that I failed at that limit.)

**Comment by jeffrey-friedl** [comment]
After failing on today's word, I really find that I'd like as many guesses as it takes to get the final answer. For stats and such, you can consider "more than 12" to be one final bucket, to simplify all that. Or maybe after failing on the 12th guess, consider it a failure but allow more guesses until one gets the right answer, or gives up. I didn't want to give up or be told the answer.... I have *some* pride 😂 to want to finish, even if I've failed.

**Comment by taximanli** [comment]
> I hate the new version... it exposes my poor Japanese skill even more than the old version. 😂
> 
> Thanks so much for your work. Three suggestions:
> 
> 1. Add "あそぼう" as an allowed guess, just because.
> 2. Have "hint mode" be off by default.
> 3. In the 「type かな here」box, and/or the guess boxes prior to the guess being officially submitted, highlight (or de-highlight?) characters that have been shown to not be in the answer.  I would find this helpful, especially with the new ability to type the guess (thanks so much for that!)
> 
> Thanks!

1. as you might already found out, I've added あそぼう as an allowed guess.
2. I have made "Hint Mode" can be turned off at the start and at the end of the game. :)
3. Good idea. This will be relatively easy to implement. I'll prioritise this in my list of future enhancements.
4. The rest of the suggestions about continuing playing the game after exhausted 12 guesses... This will take a bit more time to develop, and I need to think it through, but stay tuned!
